python/cudf/cudf/core/column/column.py

```diff
@@ -1040,7 +1040,12 @@ def serialize(self) -> Tuple[dict, list]:
         header: Dict[Any, Any] = {}
         frames = []
         header["type-serialized"] = pickle.dumps(type(self))
-        header["dtype"] = self.dtype.str
+        if hasattr(self.dtype, "str"):
+            # Notice, "dtype" must be availabe for deserialization thus
+            # if the dtype doesn't support `str` or if it is insufficient
+            # for deserialization, please overwrite the serialize and/or
+            # deserialize methods.
+            header["dtype"] = self.dtype.str
```

Comment:
Hmm, we probably need to move this implementation to `NumericalColumn.serialize` since each column type is already having their own implementation and it is just `NumericalColumn` that is using `ColumnBase.serialize`

Reply:
Same for `deserialize`

Reply:
That should also get rid of the asymmetry where `header["dtype"]` is only defined for some types during serialization, but is always expected to exist during deserialization.